packages/astro/src/types/public/config.ts

```diff
@@ -1699,6 +1700,66 @@ export interface ViteUserConfig extends OriginalViteUserConfig {
 		 * To use this feature with the Astro VS Code extension, you must also enable the `astro.content-intellisense` option in your VS Code settings. For editors using the Astro language server directly, pass the `contentIntellisense: true` initialization parameter to enable this feature.
 		 */
 		contentIntellisense?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.svg
+		 * @type {boolean|object}
+		 * @default `undefined`
+     * @version 5.x
+		 * @description
+		 * 
+		 * This feature allows you to import SVG files directly into your Astro project. By default, Astro will inline the SVG content into your HTML output.
+		 * 
+		 * To enable this feature, set `experimental.svg` to `true` in your Astro config:
+		 * 
+		 * ```js
+		 * {
+		 *   experimental: {
+		 * 	   svg: true,
+		 * 	 },
+		 * }
+		 * ```
+		 * 
+		 * To use this feature, import an SVG file in your Astro project:
+		 * 
+		 * ```astro
+		 * ---
+		 * import Logo from './path/to/svg/file.svg';
+		 * ---
+		 * 
+		 * <Logo size={24} />
+		 * ```
+		 * 
+		 * For a complete overview, and to give feedback on this experimental API,
+     * see the [Feature RFC](https://github.com/withastro/roadmap/pull/1035).
+		 */
+		svg?: {
+			/**
+			 * @docs
+			 * @name experimental.svg.mode
+			 * @type {string}
+			 * @default 'inline'
+			 * 
+			 * This mode will determine how Astro handles the imported SVG file by default. Astro will inline the SVG content into your HTML output if not specified.
```

Comment:
```suggestion
			 * The default technique for handling imported SVG files. Astro will inline the SVG content into your HTML output if not specified.
```
In my searching for content about this topic, I saw the word "technique" used a lot when comparing inline vs SVG sprites, so if you think that's a common phrase, let's go with something like that? It's nice to use a vocab familiar to the topic!

Also, we would normally show setting the config here, though it looks like it's undecided whether this will be `experimental.svg.mode` or `svg.mode`. 

I guess the maintainers prefer you make a config option outside of the experimental flag (though docs always finds it easier when the config is with the flag, because we don't have to show config in two different places, and figure out how to document a "normal" property that only exists if you've enabled a flag.... Fight, fight, fight!  :joy: )

Reply:
I think I prefer it in the `experimental` section so there is less confusion around having to also turn on the `experimental.svg` flag if someone just sees the `svg` section. 

I had made some changes to better handle the `experimental.svg` as a `boolean` and normalized it into the object. I think this will move well once stable too.

But I'm happy to do whatever the consensus is :)